src/clj/rems/workflow/dynamic.clj

```diff
@@ -529,29 +528,33 @@
   (ok {:event/type :application.event/closed
        :application/comment (:comment cmd)}))
 
+(defn- must-not-be-empty [cmd key]
+  (when-not (seq (get cmd key))
+    {:errors [{:type :must-not-be-empty :key key}]}))
+
 (defmethod command-handler ::request-decision
   [cmd _application injections]
-  (or (valid-user-error injections (:decider cmd))
+  (or (must-not-be-empty cmd :deciders)
+      (invalid-users-errors (:deciders cmd) injections)
       (ok {:event/type :application.event/decision-requested
-           :application/decider (:decider cmd)
-           :application/comment (:comment cmd)})))
-
-(defmethod command-handler ::decide
-  [cmd _application _injections]
-  (or (when-not (contains? #{:approved :rejected} (:decision cmd))
-        {:errors [{:type :invalid-decision :decision (:decision cmd)}]})
-      (ok {:event/type :application.event/decided
-           :application/decision (:decision cmd)
+           :application/request-id (java.util.UUID/randomUUID)
```

Comment:
so what was the rationale for request-id?

Reply:
There could be several comment requests going on at the same time?

Reply:
Yes. While currently we let deciders only act on the latest decision request, this was just a guess and didn't come from users. It might just as well be that deciders can reply to multiple decision requests in any order they want to.

We'll have to see, but I thought that an explicit link would make sense, and that it's a bit lighter approach to put it in the JSON blob.